wp-content/mu-plugins/pantheon/pantheon-updates.php

```diff
@@ -15,12 +16,54 @@ function _pantheon_hide_update_nag() {
 	remove_action( 'network_admin_notices', 'update_nag', 3 );
 }
 
-// Compare the current WordPress version to the latest available
+// Get the latest WordPress version
+function _pantheon_get_latest_wordpress_version() {
+	$core_updates = get_core_updates( array('dismissed' => false) );
+
+	if( ! is_array($core_updates) || empty($core_updates) || ! property_exists($core_updates[0], 'current' ) ){
+		return null;
+	}
+
+	return $core_updates[0]->current;
+}
+
+// Check if WordPress core is at the latest version.
+function _pantheon_is_wordpress_core_latest() {
+	$latest_wp_version = _pantheon_get_latest_wordpress_version();
+
+	if( null === $latest_wp_version ){
+		return true;
+	}
+
+	// include an unmodified $wp_version
+	include( ABSPATH . WPINC . '/version.php' );
+
+	// Return true if our version is the latest
+	return version_compare( str_replace( '-src', '', $latest_wp_version ), str_replace( '-src', '', $wp_version ), '=' );
+
+}
+
+// Check if the site is using the default WordPress upstream.
+function _pantheon_is_default_upstream() {
+	$data = _pantheon_curl_cached( 'https://api.live.getpantheon.com/sites/self/code-upstream-updates' );
```

Comment:
So, here is a strategy. If we hit the above API URL, then we will always get `$data['remote_url']` and `$data['remote_head']`. These entries will be the same regardless of environment, so it does not matter that the API is always returning info for the dev branch. We can just ignore the update_log and instead use:
```
git branch --contains <sha>
* master
```
If the remote_head of our upstream is a commit in the current branch, then we are up to date, and there is no upstream update.

If remote_head does not exist in the current branch, then one of two situations is the case:

1. An upstream update is needed
2. The user updated wrong (e.g. used the WordPress dashboard, squashed multiple commits together, etc).

The second scenario will cause problems in the long run (The dashboard will think that the update is still needed, the next update won't apply, etc.), so we could consider reporting the situation and pointing the user at the right documentation page.

We can test to see if the remote has a tag for the current version via:
```
git ls-remote --tags <remote-url>
```
This will return a bunch of sha hashes followed by `refs/tags/<version>`. If there is a version that is greater than the current WordPress version installed, then we know that an update is needed. Note, however, that if the site is using a custom upstream, then the tag might not be present. Also, we might get the wrong answer if there is an upstream release that is not a WordPress release, e.g. a PHP version bump in pantheon.yml, so this check is not perfect.

Reply:
Updated the API call to optionally include the branch if on a multidev.